nncf/experimental/post_training/api/engine.py

```diff
@@ -11,38 +11,128 @@
  limitations under the License.
 """
 
-from abc import ABC
 from abc import abstractmethod
 
 from typing import Dict
 from typing import TypeVar
+from typing import Callable
+from nncf.experimental.post_training.api.sampler import Sampler
 
 ModelType = TypeVar('ModelType')
 TensorType = TypeVar('TensorType')
+MetricType = TypeVar('MetricType')
 
 
-class Engine(ABC):
+class Engine:
     """
     The basic class aims to provide the interface to infer the model.
     """
-
+    # TODO (Nikita Malinin): Update class with the _get_sampler() method
     def __init__(self):
         self.model = None
+        self._sampler = None
+        self._dataset = None
+        self._metrics = None
+        self._inputs_transforms = lambda input_data: input_data
+        self._outputs_transforms = lambda output_data: output_data
+
+    # TODO (Nikita Malinin): Add statistic aggregator object (per-backend)
+    @property
+    def dataset(self):
+        return self._dataset
+
+    @dataset.setter
+    def dataset(self, dataset):
+        self._dataset = dataset
+
+    @property
+    def metrics(self):
```

Comment:
Let's use only one variant of attribute setters.
Now, we have `properties `and `set_model()`.
I suggest switching to aka `set_model()` variant as we don't suppose to have to get methods.

Reply:
Switched setters to follow `set_*()`